headerCheck/headerCheckHandler.go

```diff
@@ -0,0 +1,86 @@
+package headerCheck
+
+import (
+	"context"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go/process/headerCheck"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/disabled"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/headerCheck/factory"
+)
+
+// NewHeaderCheckHandler will create all components needed for header
+// verification and returns the header verifier component. It behaves like a
+// main factory for header verification components
+func NewHeaderCheckHandler(proxy Proxy) (HeaderVerifier, error) {
```

Comment:
this could have been placed in the factory package as this is a method factory.

Reply:
if I move it inside factory package inside headerCheck it will get into circular dependency. renamed file to `headerCheckFactory` to be more explicit; I did not create factory package outside headerCheck package since it is strongly related to it, at least for now